File: iOS/Code/Swift/bugTrap/bugTrapKit/Window/BtOverlayWindow.swift
//
//  BtOverlayWindow.swift
//  bugTrap
//
//  Created by Colby L Williams on 8/23/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation

class BtOverlayWindow: UIWindow {
	
	init() {
		super.init(frame: UIScreen.mainScreen().bounds)
		
		let btOverlayViewController = BtOverlayViewController()

		rootViewController = btOverlayViewController

		addSubview(btOverlayViewController.view)
	}
	
	required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	

	override func hitTest(point: CGPoint, withEvent event: UIEvent?) -> UIView? {

		let result = super.hitTest(point, withEvent: event)
		
		return result is BtOverlayView ? nil : result
	}
}

File: iOS/Code/Xamarin/bugTrap/bugTrapKit/OverlayWindow/BtOverlayWindow.cs
using UIKit;
using CoreGraphics;

namespace bugTrapKit
{
	public sealed class BtOverlayWindow : UIWindow
	{
		public BtOverlayWindow () : base(UIScreen.MainScreen.Bounds)
		{
			RootViewController = new BtOverlayViewController ();
			this.AddSubview(RootViewController.View);
		}

		public void BuildUp ()
		{
			var root = RootViewController as BtOverlayViewController;
			if (root != null) {
				// root.EnsureBuildUp();
			}
		}

		public override UIView HitTest (CGPoint point, UIEvent uievent)
		{
			UIView result = base.HitTest(point, uievent);

			return (result is BtOverlayView) ? null : result;
		}
	}
}